modules/javafx.controls/src/main/java/javafx/scene/control/TableView.java

```diff
@@ -2813,8 +2813,20 @@ private void quietClearSelection() {
             stopAtomic();
         }
 
-        @Override public boolean isSelected(int index) {
-            return isSelected(index, null);
+        @Override
+        public boolean isSelected(int index) {
+            final boolean isCellSelectionEnabled = isCellSelectionEnabled();
+            if (isCellSelectionEnabled) {
```

Comment:
+1

Reply:
>     1. javadoc for TableSelectionModel.isSelected(int, TablecolumnBase) already describes the logic:
Yes. I am aware of this documentation. I was thinking of making it clear for the users if they see their application breaks due to this change. I see that you have captured this in "Compatibility Impact" section of the CSR.


>     2. TreeTableView.TreeTableViewSelectionModel extends TableSelectionModel, so the changes affect both.
I see that there are overridden methods `public boolean isSelected(int index)` and `public boolean isSelected(int row, TableColumnBase<TreeItem<S>,?> column)` in class `TreeTableViewArrayListSelectionModel` present in TreeTableView.java. Hence, I think that the change that you have made in `TableView.java - isSelected(int index)` method should also be made in `TreeTableView.java - isSelected(int index)` as well.


>     3. good point, fixed.
Thanks!

Reply:
1. Would you suggest possible clarification please?  The javadoc for SelectionModel.isSelected(int) is changed to say
```
Convenience method to inform if the given index is currently selected
     * in this SelectionModel.  It will return true when at least one cell
     * is selected in the specified row index.
```

2. you are right!  fixed.

Reply:
@aghaisas 

> Yes. I am aware of this documentation. I was thinking of making it clear for the users if they see their application breaks due to this change.

If any application breaks, it's its own fault: they coded against the implementation of TableXXSelectionModels which was (incorrectly) `isSelected(int) == isSelected(int, null)`.

That might be due to the specification mess, though: 

- the old spec on `SelectionModel.isSelected(int)` was incorrectly arguing with api on MultipleSelectionModel, that is _Is functionally equivalent to calling getSelectedIndices().contains(index)_
- that invariant actually belongs into `MultipleSelectionModel.isSelected(int)` which has no specialized spec
- the method is first implemented in the (unfortunately hidden) `MultipleSelectionModelBase.isSelected(int)` fulfilling the constraint but also without spec.

To clean this up completely, we could also change the of MultipleSelectionModel and move the old `isSelected(int) == getSelectedIndices().contains(int)` into the spec of its isSelected. Probably could be done in a follow-up issue (or added to one of the open doc errors around selection models).

What do you think?

Reply:
> To clean this up completely, we could also change the of MultipleSelectionModel and move the old `isSelected(int) == getSelectedIndices().contains(int)` into the spec of its isSelected. Probably could be done in a follow-up issue (or added to one of the open doc errors around selection models).
> 
> What do you think?

I like this idea. Either filing a new bug or fold into an existing one seems fine.